### PR TITLE
aws: Add suggestion about rds_instance in rds

### LINKED ISSUE #54686: rds module in python3 environment will fail if using boto.rds2
##### SUMMARY
It's easy to reproduce, just setup rds module parameter and running ansible with python3 will trigger the problem. Change back python 2 will resolve the problem

The root cause is in this line:
boto/rds2/layer1.py 3774 line
```
body = response.read() <<<<
        boto.log.debug(body)
        if response.status == 200:
            return json.loads(body)
        else:
            json_body = json.loads(body)
```
Change this line to 
```
body = response.read().decode('utf-8')
```
Since this is boto problem and boto not maintained rds long time.
I suggest rds module should detect python3 and use boto3 as connection

The full traceback is:

Traceback (most recent call last):
  File "/home/vagrant/.ansible/tmp/ansible-tmp-1554131281.9413092-161674210350698/AnsiballZ_rds.py", line 113, in <module>
    _ansiballz_main()
  File "/home/vagrant/.ansible/tmp/ansible-tmp-1554131281.9413092-161674210350698/AnsiballZ_rds.py", line 105, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/home/vagrant/.ansible/tmp/ansible-tmp-1554131281.9413092-161674210350698/AnsiballZ_rds.py", line 48, in invoke_module
    imp.load_module('__main__', mod, module, MOD_DESC)
  File "/home/vagrant/.pyenv/versions/3.5.7/lib/python3.5/imp.py", line 235, in load_module
    return load_source(name, filename, file)
  File "/home/vagrant/.pyenv/versions/3.5.7/lib/python3.5/imp.py", line 170, in load_source
    module = _exec(spec, sys.modules[name])
  File "<frozen importlib._bootstrap>", line 626, in _exec
  File "<frozen importlib._bootstrap_external>", line 697, in exec_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
  File "/tmp/ansible_rds_payload_00xpeden/__main__.py", line 1343, in <module>
  File "/tmp/ansible_rds_payload_00xpeden/__main__.py", line 1339, in main
  File "/tmp/ansible_rds_payload_00xpeden/__main__.py", line 923, in create_db_instance
  File "/tmp/ansible_rds_payload_00xpeden/__main__.py", line 615, in get_db_instance
  File "/tmp/ansible_rds_payload_00xpeden/__main__.py", line 608, in get_db_instance
  File "/home/vagrant/.py35/lib/python3.5/site-packages/boto/rds2/layer1.py", line 1522, in describe_db_instances
    path='/', params=params)
  File "/home/vagrant/.py35/lib/python3.5/site-packages/boto/rds2/layer1.py", line 3779, in _make_request
    json_body = json.loads(body)
  File "/home/vagrant/.pyenv/versions/3.5.7/lib/python3.5/json/__init__.py", line 312, in loads
    s.__class__.__name__))
TypeError: the JSON object must be str, not 'bytes'

##### ISSUE TYPE
- Documentation Report

##### COMPONENT NAME
module name: rds

##### ANSIBLE VERSION

```
ansible 2.7.9
```

##### CONFIGURATION
```
DEFAULT_BECOME_USER(/vagrant_data/ansible-modules/ansible.cfg) = ubuntu
DEFAULT_HOST_LIST(/vagrant_data/ansible-modules/ansible.cfg) = ['/vagrant_data/ansible-modules/inventory']
DEFAULT_MODULE_PATH(/vagrant_data/ansible-modules/ansible.cfg) = ['/home/vagrant/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules', '/vagrant_data/ansible-modules/modules']
DEFAULT_REMOTE_PORT(/vagrant_data/ansible-modules/ansible.cfg) = 22
DEFAULT_REMOTE_USER(/vagrant_data/ansible-modules/ansible.cfg) = ubuntu
DEFAULT_ROLES_PATH(/vagrant_data/ansible-modules/ansible.cfg) = ['/vagrant_data/ansible-modules/roles']
HOST_KEY_CHECKING(/vagrant_data/ansible-modules/ansible.cfg) = False
```
##### OS / ENVIRONMENT
ubuntu 18.04.2 LTS

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
rds_instance is boto3 based module, add a note in rds module documentation
about the same.

Fixes: #54686

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
lib/ansible/modules/cloud/amazon/rds.py

### CODE REVIEW

**Comment by willthames** [comment]
Should we just deprecate rds and have done with it?

The last missing feature is rds_snapshot in https://github.com/ansible/ansible/pull/39994

**Comment by samccann** [comment]
From a docs perspective this looks okay, but I'd like to see the above discussion resolved about whether this should be deprecated as well.

File: lib/ansible/modules/cloud/amazon/rds.py [hunk 1]
```diff
@@ -17,9 +17,11 @@
...
 short_description: create, delete, or modify an Amazon rds instance
 description:
-     - Creates, deletes, or modifies rds instances.  When creating an instance it can be either a new instance or a read-only replica of an existing
-       instance. This module has a dependency on python-boto >= 2.5. The 'promote' command requires boto >= 2.18.0. Certain features such as tags rely
-       on boto.rds2 (boto >= 2.26.0)
+    - Creates, deletes, or modifies rds instances.
+    - When creating an instance it can be either a new instance or a read-only replica of an existing instance.
+    - This module has a dependency on python-boto >= 2.5. The 'promote' command requires boto >= 2.18.0. Certain features such as tags rely
+      on boto.rds2 (boto >= 2.26.0)
+    - Please use boto3 based M(rds_instance) instead.
```
Comment (s-hertel) [inline lib/ansible/modules/cloud/amazon/rds.py:24]: The documentation `Creates, deletes, or modifies rds instances...` wasn't really true before as the module also manages snapshots and facts. For thoroughness maybe also add `rds_snapshot`, `rds_snapshot_facts`, `rds_instance_facts` along with `rds_instance` as the modules to use instead (or in the deprecation message if not here).

I'm in favor of just deprecating this module, but let's get rds_snapshot in first.

**Comment by acozine** [comment]
+1 to deprecating the module. @s-hertel any idea of the timeframe? If we can merge the `rds_snapshot` module within days, I'll close this. Otherwise I think it makes sense to update the docs and publish the fix.

**Comment by s-hertel** [comment]
@acozine Hehe, ansibot took care of the decision for us because of the +1. The PR is linked. It looks more or less ready to merge. @willthames might have a better idea of the timeframe if he has time to finish it up.

I still think this commit could be clarified to point to the additional modules, since rds_instance isn't a complete replacement.

**Comment by willthames** [comment]
@acozine - I'll do my best in the next week (for some reason I read six days, that's probably reasonable) https://github.com/ansible/ansible/pull/39994#issuecomment-500631755

File: lib/ansible/modules/cloud/amazon/rds.py (hunk 1 above)
Comment (willthames) [reply lib/ansible/modules/cloud/amazon/rds.py:24]: Completely agree with this - suggests the need for a new PR to get the references to the other three modules in.

**Comment by acozine** [comment]
I didn't mean to make the bot merge this, but it gives us some time flexibility. @willthames thank you for your work on rds_snapshot!

**Pull request merged**